egs/librispeech/ASR/transducer_stateless/beam_search.py

```diff
@@ -348,47 +352,70 @@ def modified_beam_search(
     T = encoder_out.size(1)
 
     B = HypothesisList()
-    B.add(Hypothesis(ys=[blank_id] * context_size, log_prob=0.0))
+    B.add(
+        Hypothesis(
+            ys=[blank_id] * context_size,
+            log_prob=torch.zeros(1, dtype=torch.float32, device=device),
+        )
+    )
 
     encoder_out_len = torch.tensor([1])
     decoder_out_len = torch.tensor([1])
 
-    decoder_cache: Dict[str, torch.Tensor] = {}
     for t in range(T):
         # fmt: off
         current_encoder_out = encoder_out[:, t:t+1, :]
+        # current_encoder_out is of shape (1, 1, encoder_out_dim)
         # fmt: on
-        A = B
+        A = list(B)
         B = HypothesisList()
 
-        joint_cache: Dict[str, torch.Tensor] = {}
+        ys_log_probs = torch.cat([hyp.log_prob.reshape(1, 1) for hyp in A])
```

Comment:
With the modifier transducer we can remove the inner loop in the beam search as the maximum number of symbols per frame is 1. This simplifies the decoding process significantly. 

Also, we process all hypotheses in parallel here.